prusti-server/src/process_verification.rs

```diff
@@ -104,11 +183,21 @@ pub fn process_verification_request<'v, 't: 'v>(
         // Create a new verifier each time.
         // Workaround for https://github.com/viperproject/prusti-dev/issues/744
         let mut stopwatch = Stopwatch::start("prusti-server", "verifier startup");
+        let backend = request.backend_config.backend;
         let mut verifier =
             new_viper_verifier(program_name, verification_context, request.backend_config);
 
         stopwatch.start_next("verification");
-        let mut result = verifier.verify(viper_program);
+        let mut result = if let VerificationBackend::Lithium = backend {
+            if let prusti_common::vir::program::Program::Low(ref low_program) = request.program {
+                stopwatch.start_next("verification");
+                verifier.verify_vir(low_program)
```

Comment:
I would suggest having a `Verifier` trait that is implemented by `ViperVerifier` and `NativeVerifier`. That trait then would have a `verify` method that accepts a VIR program, which `ViperVerifier` would internally convert to a Viper program.

Reply:
All test projects expect a `verify` method that accepts a Viper program. I exposed such method on the `JavaVerifier` so that nothing needs to be renamed in test projects. The method I placed on the trait is named `verify_vir` and accepts a `Program::Legacy` or `Program::Low`, with the `JavaVerifier` transforming it to Viper before passing it to its `verify` method.

Reply:
Since the test projects are expected to construct the `JavaVerifier` directly, this sounds good to me.